lib/hyacinth/adapters/preservation_adapter/fedora3/text_resource_properties.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Hyacinth
+  module Adapters
+    module PreservationAdapter
+      class Fedora3::TextResourceProperties
+        include Fedora3::DatastreamMethods
+
+        def self.from(hyacinth_obj)
+          new(hyacinth_obj)
+        end
+
+        def initialize(hyacinth_obj)
+          @hyacinth_obj = hyacinth_obj
+        end
+
+        def to(fedora_obj)
+          return unless @hyacinth_obj.is_a? ::DigitalObject::Asset
+          ::DigitalObject::Asset::TEXT_RESOURCE_NAMES.each do |resource_name|
```

Comment:
@barmintor This implementation, only around text resources on assets, is more specific than I was imagining.  Can you make it more general so that the adapter iterates through all preservable resources on any digital object type, instead of just the ones that are hard-coded in the DigitalObject::Asset::TEXT_RESOURCE_NAMES constant?

Reply:
This is definitely the HYACINTH-501 implementation - I think the HYACINTH-814 commits will make it more general.

Reply:
That works for me. 👍